legend-engine-language-pure-compiler/src/main/java/org/finos/legend/engine/language/pure/compiler/toPureGraph/validator/MappingValidator.java

```
@@ -240,7 +248,54 @@ public void validateMappingElementIds(PureModel pureModel, Map<String, Mapping>
+
+                                                    if (ppm._transform instanceof Root_meta_pure_metamodel_function_LambdaFunction_Impl)
+                                                    {
+                                                        (((Root_meta_pure_metamodel_function_LambdaFunction_Impl)ppm._transform)._expressionSequence).forEach(ex ->
+                                                        {
+                                                            RichIterable<? extends FunctionExpression> feList = core_pure_corefunctions_metaExtension.Root_meta_pure_functions_meta_findExpressionsForFunctionInValueSpecification_ValueSpecification_1__Function_MANY__FunctionExpression_MANY_(
+                                                                            (ValueSpecification)ex,
+                                                                            Lists.mutable.of(pureModel.getFunction("meta::pure::functions::lang::new_Class_1__String_1__KeyExpression_MANY__T_1_",true)),
+                                                                            pureModel.getExecutionSupport());
+
+                                                            for (FunctionExpression fe: feList)
+                                                            {
+                                                                String errorMessage = "The new function/operator(^) is unsupported in property mappings, violated in property '" + ppm._transform + "' of mapping '" + mappingPath + "'";
+
+                                                                if (fe._genericType()._rawType() == ((ValueSpecification) ex)._genericType()._rawType())
+                                                                {
+                                                                    throw new EngineException(errorMessage, mappings.get(mappingPath).sourceInformation, EngineErrorType.COMPILATION);
```

we may want to throw a different error message in case of exception. Something which mentions that calling new on target class <className> is not allowed